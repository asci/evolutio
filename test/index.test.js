const { spawn } = require('child_process');

function getOutput(file, winners, cb) {
  const ls = spawn(
    'node',
    `./cli/index.js -p -d -s ${file} -w ${winners}`.split(' ')
  );
  let data = '';

  ls.stdout.on('data', dataPart => {
    if (!dataPart) return;

    data += dataPart;
  });

  ls.stderr.on('error', error => {
    cb(error);
  });

  ls.on('close', code => {
    // Remove time of execution
    cb(null, data.split('\n').slice(0, -2).join('\n'));
  });
}

describe('main func js', () => {
  test('it transform correctly', done => {
    getOutput('test/code/index.js', 'test/winners.json', (err, out) => {
      if (err) return done(err);
      expect(out).toMatchSnapshot();
      done();
    });
  });
});
