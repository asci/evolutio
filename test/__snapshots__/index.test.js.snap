// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main func js it transform correctly 1`] = `
"Processing 1 files... 
Spawning 1 workers...
Running in dry mode, no files will be written! 
Sending 1 files to free worker...
const evo = require('../../src');
const dna = {
  currentValue: 'a'
};

const { Mutation } = evo(dna);

function someFucntion() {
  const buttonText = 'Buy now';

  const buttonClass = 'normal';

  function extenalFunc() {
    console.log('declarative function');
  }

  extenalFunc();

  return \`<button class=\\"\${buttonClass}\\">\${buttonText}</button>\`
}

console.log(someFucntion());

All done. 
Results: 
0 errors
0 unmodified
0 skipped
1 ok"
`;
